tdrs-backend/tdpservice/search_indexes/management/commands/clean_and_reparse.py

```diff
@@ -0,0 +1,236 @@
+                    level='critical')
+                raise e
+
+            # latest version only? -> possible new ticket
+            parser_task.parse.delay(f.pk, should_send_submission_email=False)
+
+    def handle(self, *args, **options):
+        """Delete and re-parse datafiles matching a query."""
+        fiscal_year = options.get('fiscal_year', None)
+        fiscal_quarter = options.get('fiscal_quarter', None)
+        delete_all = options.get('all', False)
+        new_indices = options.get('new_indices', False)
+        delete_indices = options.get('delete_indices', False)
+
+        backup_file_name = "/tmp/reparsing_backup"
+        files = None
+        continue_msg = "You have selected to re-parse datafiles for FY {fy} and {q}. The re-parsed files "
+        if delete_all:
+            files = DataFile.objects.all()
+            backup_file_name += "_FY_All_Q1-4"
+            continue_msg = continue_msg.format(fy="All", q="Q1-4")
+        else:
+            if not fiscal_year and not fiscal_quarter:
+                print(
+                    'Options --fiscal_year and --fiscal_quarter not set. '
+                    'Provide either option to continue, or --all to wipe all submissions.'
+                )
+                return
+            files = DataFile.objects.all()
+            if (fiscal_year and fiscal_quarter):
+                files = files.filter(year=fiscal_year, quarter=fiscal_quarter)
+                backup_file_name += f"_FY_{fiscal_year}_Q{fiscal_quarter}"
+                continue_msg = continue_msg.format(fy=fiscal_year, q=fiscal_quarter)
+            elif fiscal_year:
+                files = files.filter(year=fiscal_year)
+                backup_file_name += f"_FY_{fiscal_year}_Q1-4"
+                continue_msg = continue_msg.format(fy=fiscal_year, q="Q1-4")
+            elif fiscal_quarter:
+                files = files.filter(quarter=fiscal_quarter)
+                backup_file_name += f"_FY_All_Q{fiscal_quarter}"
+                continue_msg = continue_msg.format(fy="All", q=fiscal_quarter)
+
+        continue_msg += "will {new_index} stored in new indices and the old indices ".format(new_index="be"
+                                                                                             if new_indices
+                                                                                             else "NOT be")
+        continue_msg += "will {old_index} deleted.".format(old_index="be" if delete_indices else "NOT be")
+        continue_msg += "\nThese options will delete and re-parse {0} datafiles.".format(f"ALL ({files.count()})" if
+                                                                                         delete_all else
+                                                                                         f"({files.count()})")
+
+        c = str(input(f'\n{continue_msg}\nContinue [y/n]? ')).lower()
+        if c not in ['y', 'yes']:
+            print('Cancelled.')
```

Comment:
Logger needed or line no longer needed?

Reply:
This needs to be generic std::out for the user perspective. Doesn't make a lot of sense to have a logger write this from the user perspective.